plugins/catalog-backend-module-gitlab/src/GitLabDiscoveryProcessor.ts

```diff
@@ -41,8 +41,12 @@ export class GitLabDiscoveryProcessor implements CatalogProcessor {
   private readonly integrations: ScmIntegrationRegistry;
   private readonly logger: Logger;
   private readonly cache: CacheClient;
+  private readonly checkFileExistence: boolean;
 
-  static fromConfig(config: Config, options: { logger: Logger }) {
+  static fromConfig(
+    config: Config,
+    options: { logger: Logger; checkFileExistence?: boolean },
```

Comment:
I think we can afford to be very verbose in the naming here, perhaps what freben suggested initially, `excludeReposWithoutMatchingFile`, or `eagerlyCheckFileExistence`, `skipReposWithoutExactFileMatch`?
Think I might like the last one the most as it highlights the need for an exact match